.github/workflows/main.yml

```diff
@@ -114,54 +126,75 @@ jobs:
     runs-on: ubuntu-20.04
     steps:
-    - uses: actions/checkout@v3
-
-    - name: Set up Docker Buildx
-      uses: docker/setup-buildx-action@v1
-
-
-    - name: Login to Docker Hub
-      uses: docker/login-action@v1
-      with:
-        username: ${{ secrets.DOCKERHUB_USERNAME }}
-        password: ${{ secrets.DOCKERHUB_TOKEN }}
-
-    - name: Docker meta
-      id: meta
-      uses: docker/metadata-action@v3
-      with:
-        images: baseten/truss-context-builder
-
-    - name: Docker Build
-      id: docker_build
-      uses: docker/build-push-action@v2
-      with:
-        context: ./
-        file: ./context_builder.Dockerfile
-        push: ${{ github.event_name != 'pull_request' }}
-        tags: baseten/truss-context-builder:v${{ needs.detect-version-changed.outputs.new_version }}
-        labels: ${{ steps.meta.outputs.labels }}
+      - uses: actions/checkout@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: baseten/truss-context-builder
+
+      - name: Docker Build
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          context: ./
+          file: ./context_builder.Dockerfile
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: baseten/truss-context-builder:v${{ needs.detect-version-changed.outputs.new_version }}
+          labels: ${{ steps.meta.outputs.labels }}
 
   publish-to-pypi:
     needs: [integration-tests, detect-version-changed]
-    if: ${{ !failure() && !cancelled() &&  needs.detect-version-changed.outputs.version_changed == 'true' && (needs.integration-tests.result == 'success' || needs.integration-tests.result == 'skipped') }}
+    if: ${{ !failure() && !cancelled() &&  needs.detect-version-changed.outputs.release_version == 'true' && (needs.integration-tests.result == 'success' || needs.integration-tests.result == 'skipped') }}
     runs-on: ubuntu-20.04
     steps:
-    - uses: actions/checkout@v3
+      - uses: actions/checkout@v3
 
-    - uses: ./.github/actions/setup-python/
+      - uses: ./.github/actions/setup-python/
 
-    - name: Install poetry packages
-      run: poetry install --no-dev
+      - name: Install poetry packages
+        run: poetry install --no-dev
 
-    - name: Build
-      run: poetry build
+      - name: Build
+        run: poetry build
 
-    - name: Publish to PyPI
-      if: ${{ github.event_name != 'pull_request' }}
-      run: poetry publish -u "${{ secrets.PYPI_USERNAME }}" -p "${{ secrets.PYPI_PASSWORD }}"
+      - name: Publish to PyPI
+        if: ${{ github.event_name != 'pull_request' }}
```

Comment:
might be good to hvae a comment above here explaining this logic. Are the other github events here related to merging?

Reply:
yeah. lemme refactor that part a little and add some comments. might as well get the poetry assets in the release too

Reply:
Lemme actually address this in a followup with a separate release branch.